### ISSUE TITLE
LG-235 mpHealth-2.1 implementation

### ISSUE THREAD

**Issue by donbourne** [OpenLiberty/open-liberty#8655]
Eclipse MicroProfile Health: https://github.com/eclipse/microprofile-health/tree/2.1

AHA: https://bigblue.aha.io/features/LG-235
ID: https://github.com/OpenLiberty/docs/issues/607

Target: 19.0.0.11
Feature completion: 10/16/2019

**Comment by marikaj123** [comment]
If the feature is planned to GM for the next release, please add the label target:ga.

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same BOX folder.  Thanks!

**Comment by pgunapal** [comment]
@skasund There isn't much of a difference between mpHealth-2.0 and mpHealth-2.1, as the STE for mpHealth-2.0 covers pretty much everything for mpHealth-2.1. The minor change in mpHealth-2.1, is just an additional convenient method that was added in the API to help construct the HealthCheckResponses more easier. Is an STE still needed for minor updates/changes?

**Comment by skasund** [comment]
@pgunapal  If there weren't any major changes, then an STE isn't needed.  Just make sure anything new gets documented in the KC.

**Comment by pgunapal** [comment]
@gscottj There are no addition of any new or change to any existing UI. May I get your approval on Accessibility?

@jhanders34 There are no significant change in performance for this feature. May I get your approval on performance?

@marikaj123 There are no globalization/translation required for this feature. May I get your approval for globalization?

**Comment by jhanders34** [comment]
Joe did performance comparison of mpHealth 2.1 and 2.0 and it showed performance to be the same as expected.  Giving perf approval.

**Comment by gscottj** [comment]
This feature does not add any new or change any existing user interfaces exposed to an end user. No new AVT required.  Approved.

**Comment by pgunapal** [comment]
@donbourne **Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

- Yes, the WAD includes the most likely problems the customers will see, which is usually in their own implementation of the HealthCheck interface, in their applications. Detailed messages will be logged in the messages.log. No major functional change was made for this mpHealth-2.1 feature, as everything is similar to mpHealth-2.0 feature.

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated?
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

- For mpHealth-2.1, just additional convenient methods were added in the API to help construct the HealthCheckResponses more easier. The common problem paths are similar to mpHealth-2.0, and was previously tested and demo'ed to SVT.

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?

- No SVT was done for this feature, as it only had minor changes compared to mpHealth-2.0. Tanya Kulik from the SVT team was contacted and agreed that SVT is not needed for this feature.

b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

- N/A

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.
- WAS L3: Liberty Log Analytics

**Comment by sabolo** [comment]
I have added a comment/question to the feature test summary.  Please can you address it so I can continue with the FAT review.   Many thanks.

**Comment by fmhwong** [comment]
Pull `mpHealth-2.1` from 9.0.0.11 due to a binary incompatible change in the metrics 2.1 API.  We want to GA both features at the same time.

**Comment by chirp1** [comment]
OL docs are done in priority order on the docs backlog. So, the doc updates might not be done in time for 19.0.0.12. So, add info as appropriate to the blog post.

**Comment by marikaj123** [comment]
@pgunapal - Thank you for confirming the feature did make the GM release 190012. When you can please close the feature.  Thank you.
